components/sup/src/main.rs

```diff
@@ -350,7 +350,7 @@ fn bldr_url_from_input(m: &ArgMatches) -> Option<String> {
 /// Resolve a channel. Taken from CLI args, or (failing that), a
 /// default value.
 fn channel(matches: &ArgMatches) -> ChannelIdent {
-    channel_from_input(matches).unwrap_or_default()
+    channel_from_input(matches).unwrap_or_default() // or configured_value?
```

Comment:
Judging from the docs, this should be `default`.

It's also used to resolve the channel for both loading packages as well as for the update channel for the Supervisor itself. Given that you can load things remotely, you could end up with a different package than you might otherwise expect if whoever is running the load command has their environment set up differently (than, say, the server the Supervisor is running on).

Sounds like making this listen to environment variables would be confusing at the very least 😅